### ISSUE TITLE
terraform plan: Error: Invalid function argument

### ISSUE THREAD

**Issue by ahmed-rezk-dev** [dealmore/terraform-aws-next-js#100]
I'm getting this error running `terraform plan`

Any help? 

```
Error: Invalid function argument

  on .terraform/modules/tf_next/main.tf line 4, in locals:
   4:   config_file          = jsondecode(file("${local.config_dir}/config.json"))
    |----------------
    | local.config_dir is "../.next-tf"

Invalid value for "path" parameter: no file exists at ../.next-tf/config.json;
this function works only with files that are distributed as part of the
configuration source code, so if this file will be created by a resource in
this configuration you must instead obtain this result from an attribute of
that resource.
```

**Comment by ofhouse** [comment]
Did you run `npx tf-next build` in the root of your Next.js project before running `terraform apply`?

Looks like the build output from the `tf-next` command (which is stored in `.next-tf` directory) is not present.

**Comment by ahmed-rezk-dev** [comment]
This a screenshot of my foulders [https://i.imgur.com/tlefhK8.png](url)

**Comment by ofhouse** [comment]
Looks like you are using Terraform Cloud for the deployment?
My guess is that the `.next-tf` folder is not uploaded correctly to Terraform Cloud and is therefore not present when running `terraform apply`.

Looking at your files it seems that your `.terraformignore` file is in the wrong place.
It's not very well documented but it should be located in the project root (`/.terraformignore`, where your `.prettierrc.js` is located) rather than in the terraform subfolder `/terraform/.terraformignore`.

Content of the `.terraformignore` file should also include the `.tf-next` directory as described in the [Readme](https://github.com/dealmore/terraform-aws-next-js#deployment-with-terraform-cloud).

**Comment by ahmed-rezk-dev** [comment]
Yes, I'm using Terraform Cloud for the deployment. I'm trying to make it work with AWS and GitHub actions with Terraform. Terraform Cloud is one of the requirement. If you think of another way to do it without the Terraform Cloud. Please let me know. 

I tried to add the .terraformignore in the main directory including the .tf-next. But still giving me the same error.

**Comment by ofhouse** [comment]
Interesting we also use Terraform Cloud for an project and it works fine for us.

Do you use the [VCS integration](https://www.terraform.io/docs/cloud/vcs/index.html) or do you execute `terraform apply` on the GitHub Actions instance first which triggers a remote apply in Terraform Cloud?

**Comment by ahmed-rezk-dev** [comment]
I'm actually trying to make work locally first. Then move to setup with Github actions but I will execute terraform apply on the GitHub Actions.

**Comment by ofhouse** [comment]
Okay, yes this sounds like a similar approach we are using for our own infrastructure which works fine.

Keep in mind that `terraform plan` or `terraform apply` do **not** trigger a build of Next.js by its own.
For a complete deployment you always have to execute `npx tf-next build` (which creates a Next.js build and pipes it to the `.next-tf` directory) and then `terraform plan` or `terraform apply` (which picks up some files from the previously generated `.next-tf` folder) in this exact order.

Looking at your [workflow here](https://github.com/ahmed-rezk-dev/nile-dahsboard/blob/main/.github/workflows/terraform.yml) I do not see any execution of `npx tf-next build` which may be the cause of the error.

**Comment by ahmed-rezk-dev** [comment]
I did before then I removed it again. I got the same error locally. That's why I'm trying to get it to work before pushing it again.

You are welcome to try my repo to see the error by yourself.

**Comment by ofhouse** [comment]
Checked it out locally and the apply was successful in Terraform Cloud for me.
Here are my steps for reproduction:

## Setup
- Terraform Version: 0.14.3
- Terraform Working Directory (In Terraform Cloud Settings): `/terraform`
  <img width="801" alt="Screenshot 2021-04-11 at 15 15 52" src="https://user-images.githubusercontent.com/472867/114305544-d6a9d580-9ad8-11eb-854e-d25296a68287.png">

## Reproduction steps

### 1. Git checkout
```sh
git clone git@github.com:ahmed-rezk-dev/nile-dahsboard.git
git checkout develop
```

### 2. Create ´.terraformignore` file in repo-root with the following content:
```plain
# .terraformignore
*
!terraform/
!**/.next-tf/**
```

### 3. Install dependencies & run `tf-next build`
```sh
yarn
yarn tf-next
```

### 4. Run `terraform apply` on Terraform Cloud

```
cd terraform
terraform init
terraform login

terraform apply
```

**Comment by ofhouse** [comment]
I'm going to close this issue because it has been inactive for 30 days ⏳. This helps to find and focus on the active issues.
